### ISSUE TITLE
Can't create a short lived user pool with a AWSCognitoIdentityInteractiveAuthenticationDelegate

### ISSUE THREAD

**Issue by evankyle** [aws/aws-sdk-ios#383]
Can't seem to register a user pool, assign its delegate, dealloc after all other refs are removed. If I register a user pool but do not assign the delegate dealloc on my object is called correctly.

Register User Pool
```objc
AWSCognitoIdentityUserPoolConfiguration *userPoolConfiguration = 
[ [AWSCognitoIdentityUserPoolConfiguration alloc] 
  initWithClientId:AuthCognitoIdentityUserPoolAppClientId
  clientSecret:AuthCognitoIdentityUserPoolAppClientSecret
  poolId:AuthCognitoIdentityUserPoolI
];
AWSServiceConfiguration *serviceConfigNoProvider = 
[ [AWSServiceConfiguration alloc] 
  initWithRegion:AWSRegionUSEast1 
  credentialsProvider:nil
];
[ AWSCognitoIdentityUserPool
  registerCognitoIdentityUserPoolWithConfiguration:serviceConfigNoProvider
  userPoolConfiguration:userPoolConfiguration
  forKey:AuthCognitoIdentityUserPoolDoPrompt
];
self.pool =
[ AWSCognitoIdentityUserPool
  CognitoIdentityUserPoolForKey:AuthCognitoIdentityUserPoolDoPrompt
];
```
Assign delegate
```objc
self.pool.delegate = self
```
remove / un-register user pool
```objc
 [AWSCognitoIdentityUserPool removeCognitoIdentityUserPoolForKey:AuthCognitoIdentityUserPoolDoPrompt];
```
Remove refs to delegate but dealloc not called.

**Comment by behrooziAWS** [comment]
A strong reference to the delegate is being kept by the user pool, we will set this to nil during dealloc to allow it to be released in the next release.

**Comment by behrooziAWS** [comment]
This has been fixed in 2.4.2.
